File: crates/domain-scan-core/tests/fixtures/scala/classes.scala
package com.example

class UserService(repository: UserRepository) {
  def getUser(id: String): Option[User] = {
    repository.findById(id)
  }

  def createUser(name: String, email: String): User = {
    val user = User(name, email)
    repository.save(user)
    user
  }

  private def validateEmail(email: String): Boolean = {
    email.contains("@")
  }
}

abstract class BaseEntity {
  def id: String
  def getType(): String
}

case class User(name: String, email: String, age: Int = 0)

case class CreateUserRequest(
  name: String,
  email: String,
  password: String
)

object UserService {
  def apply(repo: UserRepository): UserService = new UserService(repo)
}

File: crates/domain-scan-core/tests/fixtures/csharp/classes.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace DomainScan.Tests
{
    public class UserService
    {
        private readonly IUserRepository _repository;
        private readonly INotificationService _notificationService;

        public UserService(IUserRepository repository, INotificationService notificationService)
        {
            _repository = repository;
            _notificationService = notificationService;
        }

        public async Task<User> GetUser(int id)
        {
            return await _repository.GetById(id);
        }

        public async Task CreateUser(string name, string email)
        {
            var user = new User { Name = name, Email = email };
            await _repository.Save(user);
            _notificationService.SendEmail(email, "Welcome", "Hello!");
        }

        private void ValidateUser(User user)
        {
            // validation logic
        }

        public static UserService Create()
        {
            return null;
        }
    }

    public abstract class BaseEntity
    {
        public int Id { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }

    public class GenericHandler<T> where T : class
    {
        public void Handle(T item)
        {
            // handle
        }
    }

    internal class InternalHelper
    {
        public void DoWork()
        {
        }
    }
}
